### PR TITLE
[14.0] mig project_gtd

### PR DESCRIPTION
**Pull request by fredzamoabg**
**project_gtd** module migration from version 10.0

### CODE REVIEW

File: project_gtd/README.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+   :local:
+
+Usage
+=====
+
+To use this module, you need to:
+
+#. Go to `Project / Configuration / GTD / Timeboxes` to configure your timeboxes.
+#. Here, your new timebox is empty and not visible yet. You need to add tasks in it to make it visible.
+#. Go to `Project / Tasks` and edit a task to set the 'Timebox' field.
```
Comment (tafaRU) [inline project_gtd/README.rst]: From ui point of view it has label `Timeframe`:

image

I would align the labels. What do you think?

File: project_gtd/wizard/project_gtd_empty.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    _description = "Project Timebox Empty"
+
+    state = fields.Selection(
+        [("init", "init"), ("done", "done")],
+        readonly=True,
+        default="init",
+    )
+
+    def process(self):
+        close = []
```
Comment (tafaRU) [inline project_gtd/wizard/project_gtd_empty.py:20]: Could you please documenting a bit  the method by using a docstring? Not totally clear what `up` is used for?

File: project_gtd/wizard/project_gtd_fill.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    )
+    timebox_to_id = fields.Many2one(
+        "project.gtd.timebox", "Set to Timebox", required=True, default=_get_to_tb
+    )
+    task_ids = fields.Many2many(
+        "project.task", "project_task_rel", "task_id", "fill_id", "Tasks selection"
+    )
+
+    def process(self):
+        self.ensure_one()
```
Comment (tafaRU) [inline project_gtd/wizard/project_gtd_fill.py]: It is deprecated: could you please remove it?

File: project_gtd/tests/test_project_gtd.py [hunk 4]
```diff
@@ -8,6 +8,9 @@
 class TestProjectGtd(TransactionCase):
     def setUp(self, *args, **kwargs):
```
Comment (tafaRU) [inline project_gtd/tests/test_project_gtd.py]: `args` and `kwargs` are not needed: could you remove them?

File: project_gtd/tests/test_project_gtd.py [hunk 5]
```diff
@@ -42,16 +45,33 @@ def test_timebox_weekly(self):
...
+                self.timebox_weekly.id,
+            ],
+            "active_id": self.timebox_weekly.id,
+        }
+        self.project_timebox_empty.with_context(context).process()
+        self.assertFalse(self.task.timebox_id)
+
+    def test_timebox_empty_up(self):
+        """
+        Check if timebox is empty after emptiying task.
```
Comment (tafaRU) [inline project_gtd/tests/test_project_gtd.py]: It is intended this docstring is equal to the previous one?
Also, could you please avoid code repetition?

**Review by tafaRU — CHANGES_REQUESTED** [review]

File: project_gtd/README.rst (hunk 1 above)
Comment (fredzamoabg) [reply project_gtd/README.rst]: ok, I'll align labels. thanks

File: project_gtd/wizard/project_gtd_empty.py (hunk 2 above)
Comment (fredzamoabg) [reply project_gtd/wizard/project_gtd_empty.py:20]: this is just a migration of the module, I did not go into the details of the implementation.

File: project_gtd/wizard/project_gtd_fill.py (hunk 3 above)
Comment (fredzamoabg) [reply project_gtd/wizard/project_gtd_fill.py]: ok, I'll do that. thanks

File: project_gtd/tests/test_project_gtd.py (hunk 4 above)
Comment (fredzamoabg) [reply project_gtd/tests/test_project_gtd.py]: ok, I'll do that. thanks

File: project_gtd/tests/test_project_gtd.py (hunk 5 above)
Comment (fredzamoabg) [reply project_gtd/tests/test_project_gtd.py]: I'll check, thanks

Comment (tafaRU) [reply project_gtd/tests/test_project_gtd.py]: This docstring seems not right yet. Could you please check it?

Comment (fredzamoabg) [reply project_gtd/tests/test_project_gtd.py]: done

**Review by tafaRU — APPROVED** [review]

**Review by bealdav — APPROVED** [review]
It seems ok for me, thanks

File: project_gtd/models/project_gtd.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+    @api.model
+    def fields_view_get(self, *args, **kwargs):
+        res = super(ProjectTask, self).fields_view_get(*args, **kwargs)
+        timebox_model = self.env["project.gtd.timebox"]
+        if (res["type"] == "search") and self.env.context.get("gtd", False):
+            timeboxes = timebox_model.search([])
+            search_extended = ""
+            for timebox in timeboxes:
+                filter_ = u"""
```
Comment (leemannd) [inline project_gtd/models/project_gtd.py:92]:
```suggestion
                filter_ = """
```
Hello @fredzamoabg , the newest version of the pre-commit is preventing the merge.
Could you have look so we can merge it?

**Comment by leemannd** [comment]
As we had no answer from the author, closing in favor of https://github.com/OCA/project/pull/1038

**Pull request closed without merging**